### PR TITLE
[App Service] Fix #17507: `az staticwebapp functions`: Add support for linking existing function app to static webapp (bring your own functions)

### LINKED ISSUE #17507: WebApp:staticwebapp - [P2] add support for linking existing function app (bring your own functions)
**Is your feature request related to a problem? Please describe.**

Azure Static Web Apps is adding the ability to link a Static Web App to an existing Azure Functions app. This requires gestures in Az CLI to add and remove the link.

**Describe the solution you'd like**

Link:

```bash
az staticwebapp functions link -g $StaticWebAppGroup -n $StaticWebAppName --function-resource-id $FunctionsResourceId
```

Unlink:

```bash
az staticwebapp functions unlink -g $StaticWebAppGroup -n $StaticWebAppName
```

**Additional context**

Needed for Static Web Apps GA

### PR DESCRIPTION
**Pull request by StrawnSC**
**Description**

Adds commands to support linking a preexisting Azure Function to a static webapp (aka Bring your own Functions) 

Commands added: 
`az staticwebapp functions link` 
`az staticwebapp functions unlink` 
`az staticwebapp functions show`

Requested by https://github.com/Azure/azure-cli/issues/17507

**Testing Guide**

Mocked tests: 
`azdev test test_functions_unlink test_functions_link test_functions_show`

Commands to test live: 
`az staticwebapp functions link -g ... -n ... --function-resource-id ...`
`az staticwebapp functions unlink -g ... -n ...`
`az staticwebapp functions show -g ... -n ...`

Note that there aren't automated live tests because creating a static webapp requires a GitHub auth token 

**History Notes**

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [X] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [X] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [X] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
App Service

**Review by wangzelin007 — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/appservice/_validators.py [hunk 1]
```diff
@@ -250,3 +251,29 @@ def validate_public_cloud(cmd):
...
+
+def validate_staticsite_sku(cmd, namespace):
+    from azure.mgmt.web import WebSiteManagementClient
+    client = get_mgmt_service_client(cmd.cli_ctx, WebSiteManagementClient).static_sites
+    sku_name = client.get_static_site(namespace.resource_group_name, namespace.name).sku.name
+    if sku_name.lower() != "standard":
+        raise ValidationError("Invalid SKU: '{}'. Staticwebapp must have 'Standard' SKU".format(sku_name))
+
+
+def validate_staticsite_link_function(cmd, namespace):
```
[the remark this answers is not in the document]
Comment (StrawnSC) [reply src/azure-cli/azure/cli/command_modules/appservice/_validators.py:264]: The validator `validate_staticsite_link_function` is used [here](https://github.com/Azure/azure-cli/pull/20116/files#diff-c4850c36aa0fa1257ca6eb293cd1d9775fc057de8b8bc027c258310d51c5ebedR467) as the validator for the command `az staticwebapp functions link` in the appservice commands file

[the remark this answers is not in the document]
Comment (wangzelin007) [reply src/azure-cli/azure/cli/command_modules/appservice/_validators.py:264]: Yes I found you added in commit 60fac2 later.

**Review by panchagnula — APPROVED** [review]
LGTM

**Comment by zhoxing-ms** [comment]
＠StrawnSC  Could you please address those conflicts?

**Comment by zhoxing-ms** [comment]
By the way, may I ask could you add scenario tests for these new commands?

**Comment by StrawnSC** [comment]
> By the way, may I ask could you add scenario tests for these new commands?

@zhoxing-ms like I mentioned [here](https://github.com/Azure/azure-cli/pull/20140#issuecomment-963692426), we don't have end-to-end scenario tests for the Static Web App commands because creating a Static Web App via the CLI requires passing in a GitHub auth token and connecting the SWA to a GitHub repo, unfortunately. I added a mocked test in the `test_staticapp_commands_thru_mock.py` file, however.

**Comment by zhoxing-ms** [comment]
@StrawnSC Could you please fix those conflicts?

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**